BSP/Simulator/Src/BSP_Timer.c

```diff
@@ -0,0 +1,62 @@
+#include "BSP_Timer.h"
+
+static void (*Timer1Handler)(int);
+static void (*Timer2Handler)(int);
+//had to pass some sort of parameter to the pointer or it doesn't work
+
+/** 
+* @brief   Updates the time by reading the CSV file and calling relevant functions at the right time 
+* @param   none
+* @return  none
+*/  
+void BSP_Timer_Update()
+{
+    FILE *file = fopen("Timer.csv", "r");
+    int fno = fileno(file);
+    flock(fno, LOCK_EX);
+    int timeCurrent[2];
+    int current;
+    int reload;
+    int i=0;
+    int index=0;
+    
+    while(i<2)
+    {
+        fscanf(file,"%d , %d",&current,&reload);
+        timeCurrent[index]=current;

```

Comment:
Instead of this, you could get rid of `current` altogether and simply pass (timeCurrent+index) where you have &current

Reply:
When I try to pass the array with the index value into the fscanf  function, it gives me an error when the code is compiling. It says that the right format is not being passed into the function. It works so far when I pass the address of a normal int variable